components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageTxImpl.java

```diff
@@ -409,26 +424,42 @@ public void deleteBucket(Bucket bucket) {
 
   @Override
   @Guarded(by = OPEN)
-  public BlobRef createBlob(final InputStream inputStream, Map<String, String> headers) {
+  public AssetBlob createBlob(final String blobNameHint,
```

Comment:
Since we're storing this value as the blob name, shouldn't the parameter just be called `blobName` - the use of "hint" makes me thing this is optional or trivial, whereas as @mrprescott says it does have a significance.